run/markdown-preview/editor/render.go

```diff
@@ -46,14 +50,24 @@ func (s *RenderService) NewRequest(method string) (*http.Request, error) {
 		return req, nil
 	}
 
-	// Query the id_token with ?audience as the serviceURL
-	tokenURL := fmt.Sprintf("/instance/service-accounts/default/identity?audience=%s", s.URL)
-	token, err := metadata.Get(tokenURL)
-	if err != nil {
-		return req, fmt.Errorf("metadata.Get: failed to query id_token: %w", err)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Create a TokenSource if none exists.
+	if s.tokenSource == nil {
+		tokenSource, err := idtoken.NewTokenSource(ctx, s.URL)
+		if err != nil {
+			return nil, fmt.Errorf("idtoken.NewTokenSource: %w", err)
+		}
+		// Create a caching token source to reuse tokens until expiration.
+		s.tokenSource = oauth2.ReuseTokenSource(nil, tokenSource)
 	}
 
-	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	token, err := s.tokenSource.Token()
+	if err != nil {
+		return nil, fmt.Errorf("TokenSource.Token: %w", err)
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken))
```

Comment:
Maybe we should use [SetAuthHeader](https://pkg.go.dev/golang.org/x/oauth2?tab=doc#Token.SetAuthHeader) instead?